crates/polars-ops/src/series/ops/rank.rs

```diff
@@ -42,8 +43,29 @@ fn get_random_seed() -> u64 {
     rng.next_u64()
 }
 
+unsafe fn rank_impl<F: FnMut(&mut [IdxSize])>(idxs: &IdxCa, neq: &BooleanArray, mut flush_ties: F) {
+    let mut ties_indices = Vec::with_capacity(128);
+    let mut idx_it = idxs.downcast_iter().flat_map(|arr| arr.values_iter());
```

Comment:
Can we use a double loop here, so the iteration is slightly faster?

Reply:
I think this compiles down to a double loop.